lib/Castle/Request.php

```diff
@@ -27,7 +27,13 @@ public function handleApiError($response, $status)
       case 404:
         throw new Castle_NotFoundError($msg, $type, $status);
       case 422:
-        throw new Castle_InvalidParametersError($msg, $type, $status);
+        // Handle subtype errors
+        switch($type) {
```

Comment:
I would move this to a separate method already so next person won't just add another if here.

Reply:
This can be re-factored once we add more subtypes (if ever)

Reply:
Yeah I guess. But if someone will another if here then already big cyclomatic complexity will be huge ;)